pkg/envoy/lds/inmesh.go

```diff
@@ -164,6 +171,12 @@ func (lb *listenerBuilder) getInboundMeshTCPFilterChain(proxyService service.Mes
 		return nil, err
 	}
 
+	serverNames, err := lb.meshCatalog.GetServiceHostnames(proxyService, service.RemoteCluster)
+	if err != nil {
+		// this error only affects multi cluster, so we don't return early here.
+		log.Error().Err(err).Msgf("Error fetching remote cluster service names for %s", proxyService)
+	}
```

Comment:
Please wrap this change with `if proxy.Kind == gateway`. We don't want to be exercising multicluster related code for inmesh app<-->app connectivity.

Reply:
this actually is for the non-gateways, so that they can get the correct reference. Added a comment, but also wrapped in the ENabledMultiCluster feature flag